### DISCUSSION TITLE
Sandbox for DIGIT

### DISCUSSION THREAD

**Discussion by rohitdega** [egovernments/Digit-Core#102] [Platform Projects]
**Background :** 
One of the key promises of DIGIT is to enable the community / partners to solve for new usecases. In order to accelerate participation from partners and the ecosystem, it is proposed to setup a sandbox with controlled access that can be used to test, tryout, experience and experiment on DIGIT. 

**Key features being considered under the sandbox :** 
1. Access to all product demos 
2. Enable developers/users to access the APIs to experiment - consume APIs of existing services on DIGIT
3. Enable new services to be built and deployed on the DIGIT sandbox 

**Proposed features of the sandbox :** 
- All users have open access to DIGIT documentation 
- Users signup and get a full fledged access to the guided demo of the reference applications 
- Developers who want to experiment with/on the platform, will have to make a request for “dev-access” 
- Requests auto approved and auth keys sent to the developers 
- Developers can proceed to consume APIs, make changes to MDMS, UI etc 
- Developers who want to make changes to / build new services - make a request 
- These requests will be manually approved and access enabled for developers
- Developers can modify, build, deploy and test services 
- System flushes the sandbox in regular intervals to keep usage in check - DB, access keys, users etc 
- Automated and manual protocols to prevent and control over-usage/abuse - infra, data, sys performance (time of access, bandwidth and infra consumption etc) 

Skills/Interests : 
- DevOps enthusiasts 
- Full stack engineers 
- Platform product managers 
- Prior experience building sandbox environment would be an advantage

**Comment by ghost** [comment]
This would be a very welcome development
